### DISCUSSION TITLE
Negative training ?¿?

### DISCUSSION THREAD

**Discussion by Trebol-a** [ultralytics/ultralytics#9557] [General]
Hi, people,
I am training with YOLOv8 a system for bird species identification on webcams/trailcam. The results are starting to be quite promising but still some difficult species resist it and it confuses them (they differ in small nuances or minuscule details). 
If I understand fine how "background" images work, could I teach the system that an image DOES NOT correspond to a species by leaving its TXT labelling empty, or some another way to teach it "negatively": this photo IS NOT of this species!

Thank you very much.

**Comment by Trebol-a** [comment]
![image](https://github.com/ultralytics/ultralytics/assets/20790138/a1216f54-86d5-4917-b0fc-2b84cb8b53e5)
By example, in picture, species 1,2,3 are well identified, but not 4, that is a species I have not training (and classifies like 1).
Besides adding more pictures of 1,2 and 3 is there any way to tell it 4 is not 1?

**Reply by glenn-jocher** [reply]
Hello! 😊 Great to hear about your progress with bird species identification using YOLOv8.

For distinguishing difficult species, especially to teach the model that certain images do not belong to any of the known classes, you can utilize background or negative images. Including images without any bounding box annotations (empty `.txt` files) in your training dataset effectively tells the model, "This image contains none of the objects of interest."

However, for specifically teaching it that species 4 is not species 1 (as per your example), it might be beneficial to introduce a distinct class for species 4 (if it's significant for your application) or ensure it's well-represented in the "background" class. This way, your model learns to distinguish it from the species of interest.

Additionally, improving your model further could involve tweaking data augmentation strategies or even manual feature engineering to emphasize the nuanced differences between similar species.

Keep up the good work, and best of luck with refining your model! If you need more detailed advice or examples, feel free to ask.

**Comment by Trebol-a** [comment]
Thank you very much for your comments and encouragement.
Adding a new species every time the engine confuses an untrained species is not an option. My dataset contains 70 species and 17.000 photos ... if in the world there are 10.000 species ... too much for only one life. :D
Therefore, the most "realistic" option would be to add photos of "species-to-ignore" with .txt file but without tagging.
Or in other words, if I insert an image like this in which I identify only the species I'm interested in, I'm saying "don't look at the others, ignore them"? Or for "detraining" to be most effective, there should be no recognisable species at all?
![image](https://github.com/ultralytics/ultralytics/assets/20790138/f5d3baa6-a74c-4e83-885a-9d63e713fbf8)

**Reply by glenn-jocher** [reply]
@Trebol-a hello! 😊 It's great to see your enthusiasm and how you're tackling this challenge. 

Indeed, the approach you've mentioned about using images with `.txt` files but without labels can serve as "negative" examples. By including these in your dataset, you're essentially informing the model that these images do not contain any objects of interest. This helps in refining the model's understanding of what not to detect, thereby improving its focus on the species you're interested in.

Regarding images with both target and non-target species, if you label only the species of interest and leave others unlabeled, the model learns to recognize and label only those species you've annotated. The unannotated species in the image would be treated as part of the background or "ignored." This can indirectly tell the model to "ignore" or not detect unlabelled species, although it's more about teaching the model what you're interested in rather than a direct instruction to disregard specific objects.

Here's a small tip: Using a diverse set of backgrounds and contexts for your "negative" images can enhance the model's ability to generalize well and focus on the species of interest.

Best of luck with your project, and feel free to reach out if you have more questions!

**Reply by Trebol-a** [reply]
Perfect, now everything is much clearer. 
Thank you very much for your help.

Best regards

**Reply by glenn-jocher** [reply]
You're welcome! 😊 Glad to hear everything is clearer now. If you have any more questions in the future, don't hesitate to ask. 

Best regards!

**Reply by jugal-sheth** [reply]
Hello @glenn-jocher ,

I hope this message finds you well. I am currently working on a project involving object detection using the YOLO V8 algorithm.

I have encountered a challenge in my training process that I believe your expertise could help me address. Specifically, I am working with a dataset of over 200,000 annotated images. However, I have noticed that some images contain objects that are not annotated, considering the algorithm's tendency to focus on learning about the background also along with detecting all objects present in the images. I am reaching out to understand best practices or strategies for handling this type of dataset. Given the scale of my dataset, manual annotation of the remaining objects is not feasible, so I am seeking alternative approaches to improve the model's performance.

Your insights and guidance on this matter would be greatly appreciated. If you are available for a discussion or have any resources or recommendations you could share, I would be grateful for the opportunity to learn from your expertise.

Thank you very much for your time

Warm regards,
Jugal Sheth
[Shethjugal123@gmail.com]

**Reply by glenn-jocher** [reply]
Hello Jugal,

I'm glad to hear about your project and the work you're doing with YOLOv8! 🚀

Dealing with large datasets where not all objects are annotated is quite common. A practical approach could be to use these images as "negative samples" by not including annotations for the unannotated objects. This way, your model learns these are part of the background or simply not the objects of interest. It's also a good practice to have a balanced dataset with a mix of such negative samples and fully annotated images to improve generalization.

For managing large datasets, you might find automated annotation tools helpful to preprocess or validate your data, though manual checks are recommended for a subset to ensure quality.

Lastly, if the presence of unannotated objects consistently detracts from your model's performance, consider semi-supervised learning techniques or data augmentation strategies to emphasize the features of annotated objects.

Here's a snippet on how to include such images with empty annotation files in your dataset:
```plaintext
# For an unannotated image "example.jpg", simply include an empty "example.txt" file.
```

Feel free to explore more on dataset preparation in our [Ultralytics documentation](https://docs.ultralytics.com/) for additional strategies.

Best of luck with your project, and don't hesitate to reach out for further assistance!

Warm
